lib/classes/ACL.php

```diff
@@ -1517,43 +1475,28 @@ public function duplicate_right_from_bas($base_id_from, $base_id_dest)
 
         $this->give_access_to_base([$base_id_dest]);
 
-        $rights = [
-            'mask_and' => $row['mask_and'],
-            'mask_xor' => $row['mask_xor'],
-        ];
-
-        if ($row['canputinalbum'])
-            $rights['canputinalbum'] = true;
-        if ($row['candwnldhd'])
-            $rights['candwnldhd'] = true;
-        if ($row['candwnldpreview'])
-            $rights['candwnldpreview'] = true;
-        if ($row['cancmd'])
-            $rights['cancmd'] = true;
-        if ($row['canadmin'])
-            $rights['canadmin'] = true;
-        if ($row['canreport'])
-            $rights['canreport'] = true;
-        if ($row['canpush'])
-            $rights['canpush'] = true;
-        if ($row['nowatermark'])
-            $rights['nowatermark'] = true;
-        if ($row['canaddrecord'])
-            $rights['canaddrecord'] = true;
-        if ($row['canmodifrecord'])
-            $rights['canmodifrecord'] = true;
-        if ($row['candeleterecord'])
-            $rights['candeleterecord'] = true;
-        if ($row['chgstatus'])
-            $rights['chgstatus'] = true;
-        if ($row['imgtools'])
-            $rights['imgtools'] = true;
-        if ($row['manage'])
-            $rights['manage'] = true;
-        if ($row['modify_struct'])
-            $rights['modify_struct'] = true;
-
-        $this->update_rights_to_base($base_id_dest, $rights);
+        $this->update_rights_to_base(
+            $base_id_dest,
+            [
+                'mask_and'               => $row['mask_and'],
+                'mask_xor'               => $row['mask_xor'],
+                self::CANPUTINALBUM      => ($row[self::CANPUTINALBUM] == '1'),
```

Comment:
It would be better to use a (bool) cast to make it more obvious, ie

`(bool) $row[self::CANPUTINALBUM]`